crm-project/src/main/java/CRM/service/UserService.java

```diff
@@ -140,50 +143,64 @@ public List<User> getAllInBoard(long boardId) throws AccountNotFoundException {
         }
 
-        NotificationSetting notificationSetting = Validations.doesIdExists(2L, settingRepository);
-
-        UserPermission userPermission = new UserPermission();
-        userPermission.setId(0L);
-        userPermission.setUser(user);
-        userPermission.setPermission(Permission.USER);
+        if(user.equals(board.getCreatorUser())){
+            throw new IllegalArgumentException(ExceptionMessage.ADMIN_CANT_CHANGE_HIS_PERMISSION.toString());
+        }
 
-        board.addUserPermissionToBoard(userPermission);
+        Permission permissionRequest = Permission.values()[Math.toIntExact(objectsIdsRequest.getPermissionId())];
+        Set<UserPermission> userPermissionsSet = updateUserPermission(user, permissionRequest, board);
+        List<User> users = board.getAllUsersInBoard(board, userPermissionsSet);
 
-        createDefaultSettingForNewUserInBoard(user, board, notificationSetting);
         boardRepository.save(board);
+        return users;
     }
 
-    /**
-     * Creates default notifications for every new user in every board,
-     * using constant notifications
-     */
-    private void createDefaultSettingForNewUserInBoard(User user, Board board, NotificationSetting notificationSetting) {
-        UserSetting userSetting = new UserSetting();
-        userSetting.setId(0L);
-        userSetting.setInApp(true);
-        userSetting.setInEmail(true);
-        userSetting.setUser(user);
-        userSetting.setSetting(notificationSetting);
-        board.addUserSettingToBoard(userSetting);
+    //TODO documentation
+    private void createDefaultSettingForNewUserInBoard(User user, Board board) {
+        for (long i = 0; i < 6; i++) {
+            NotificationSetting notificationSetting = Validations.doesIdExists(i+1L, settingRepository);
+            UserSetting userSetting = UserSetting.createUserSetting(user, notificationSetting);
+            board.addUserSettingToBoard(userSetting);
+        }
+    }
+
+    //TODO documentation
+    private void createNewUserPermission(User user, Permission permission, Board board){
+        UserPermission userPermission = UserPermission.newUserPermission(user, permission);
+        board.addUserPermissionToBoard(userPermission);
+        createDefaultSettingForNewUserInBoard(user, board);
+    }
+
+    //TODO documentation
+    private Set<UserPermission> updateUserPermission(User user, Permission permissionRequest, Board board) {
+        Set<UserPermission> userPermissionsSet = board.getUsersPermissions();
+        UserPermission userPermissionInBoard = board.getUserPermissionById(board, user.getId() ,userPermissionsSet);
+
+        if (permissionRequest.equals(Permission.ADMIN)) {
+            throw new IllegalArgumentException(ExceptionMessage.PERMISSION_NOT_ALLOWED.toString());
+        } else if (permissionRequest.equals(Permission.UNAUTHORIZED)) {
+            if (userPermissionInBoard == null) {
```

Comment:
Seeing it now, you don't have to check if it is null. 
Since you are using a Set, use its built-in removeIf() function:
``
set.removeIf(user -> user.id == id)
``